src/pybind/mgr/drain_osd/module.py

```diff
@@ -0,0 +1,333 @@
+                           opt['name'], getattr(self, opt['name']))
+        # Do the same for the native options.
+        for opt in self.NATIVE_OPTIONS:
+            setattr(self,
+                    opt,
+                    self.get_ceph_option(opt))
+            self.log.debug(' native option %s = %s', opt, getattr(self, opt))
+
+    def handle_command(self, inbuf, cmd):
+        ret = 0
+        err = ''
+        _ = inbuf
+        if cmd.get('prefix', '') == 'osd drain':
+            osd_ids = cmd.get('osd_ids', list())
+
+            not_found: Set[int] = self.osds_not_in_cluster(osd_ids)
+            if not_found:
+                return -errno.EINVAL, '', f"OSDs <{not_found}> not found in cluster"
+            # add osd_ids to set
+            [self.osd_ids.add(osd_id) for osd_id in osd_ids]
+
+            self.log.info(f'Scheduling OSD(s) <{self.osd_ids}> to be drained')
+
+            out = 'Started draining OSDs. Query progress with <ceph drain status>'
+
+        elif cmd.get('prefix') == 'osd drain status':
+            # re-initialize it with an empty set on invocation (long running processes)
+            self.check_osds = set()
+            # assemble a set of emptying osds and to_be_emptied osds
+            self.check_osds.update(self.emptying_osds)
+            self.check_osds.update(self.osd_ids)
+
+            report = list()
+            for osd_id in self.check_osds:
+                pgs = self.get_pg_count(osd_id)
+                report.append(dict(osd_id=osd_id, pgs=pgs))
+            out = f"{report}"
+
+        elif cmd.get('prefix', '') == 'osd drain stop':
+            osd_ids: List[int] = cmd.get('osd_ids', list())
+            if not osd_ids:
+                self.log.debug("No osd_ids provided, stop all pending drain operations)")
+                self.osd_ids = set()
+                self.emptying_osds = set()
+
+                # this is just a poor-man's solution as it will not really stop draining
+                # the osds. It will just stop the queries and also prevent any scheduled OSDs
+                # from getting drained at a later point in time.
+                out = "Stopped all future draining operations (not resetting the weight for already reweighted OSDs)"
+
+            else:
+                not_found: Set[int] = self.osds_not_in_cluster(osd_ids)
+                if not_found:
+                    return -errno.EINVAL, '', f"OSDs <{not_found}> not found in cluster"
+
+                self.osd_ids = self.osd_ids.difference(osd_ids)
+                self.emptying_osds = self.emptying_osds.difference(osd_ids)
+                out = f"Stopped draining operations for OSD(s): {osd_ids}"
+
+        else:
+            return (-errno.EINVAL, '',
+                    "Command not found '{0}'".format(cmd['prefix']))
+
+        return HandleCommandResult(
+            retval=ret,   # exit code
+            stdout=out,   # stdout
+            stderr=err)
```

Comment:
Hm. Either you use `HandleCommandResult` or you return a bare tuple. :-1: for mixing them.

Reply:
Huh, I took that from the `mgr/hello` example.. Let me adjust it